Sources/Starknet/Accounts/StarknetAccountProtocol.swift

```diff
@@ -80,98 +133,292 @@ public protocol StarknetAccountProtocol {
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV3(calls: [StarknetCall], nonce: Felt) async throws -> StarknetFeeEstimate {
+        try await estimateFeeV3(calls: calls, nonce: nonce, simulationFlags: defaultSimulationFlagsForEstimateFee)
+    }
+
+    /// Estimate fee for a list of calls as invoke transaction v1
+    ///
+    /// - Parameters:
+    ///  - calls: list of calls, for which the fee should be estimated.
+    ///  - simulationFlags: set of simulation flags for fee estimation.
+
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV1(calls: [StarknetCall], simulationFlags: Set<StarknetSimulationFlagForEstimateFee>) async throws -> StarknetFeeEstimate {
+        let nonce = try await getNonce()
+        return try await estimateFeeV1(calls: calls, nonce: nonce, simulationFlags: simulationFlags)
     }
 
-    /// Estimate fee for a list of calls
+    /// Estimate fee for a list of calls as invoke transaction v3
     ///
     /// - Parameters:
     ///  - calls: list of calls, for which the fee should be estimated.
+    ///  - simulationFlags: set of simulation flags for fee estimation.
+
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV3(calls: [StarknetCall], simulationFlags: Set<StarknetSimulationFlagForEstimateFee>) async throws -> StarknetFeeEstimate {
+        let nonce = try await getNonce()
+        return try await estimateFeeV3(calls: calls, nonce: nonce, simulationFlags: simulationFlags)
+    }
+
+    /// Estimate fee for a list of calls as invoke transaction v1
+    ///
+    /// - Parameters:
+    ///  - calls: list of calls, for which the fee should be estimated.
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV1(calls: [StarknetCall]) async throws -> StarknetFeeEstimate {
+        let nonce = try await getNonce()
+        return try await estimateFeeV1(calls: calls, nonce: nonce)
+    }
+
+    /// Estimate fee for a list of calls as invoke transaction v3
+    ///
+    /// - Parameters:
+    ///  - calls: list of calls, for which the fee should be estimated.
+    ///
     /// - Returns: struct containing fee estimate
-    func estimateFee(calls: [StarknetCall]) async throws -> StarknetFeeEstimate {
+    func estimateFeeV3(calls: [StarknetCall]) async throws -> StarknetFeeEstimate {
         let nonce = try await getNonce()
-        return try await estimateFee(calls: calls, nonce: nonce)
+        return try await estimateFeeV3(calls: calls, nonce: nonce)
+    }
+
+    /// Estimate fee for a call as invoke transaction v1
+    ///
+    /// - Parameters:
+    ///  - call: a call for which the fee should be estimated.
+    ///  - nonce: a nonce to be used in a transaction.
+    ///  - simulationFlags: a set of simulation flags.
+    ///
+    /// - Returns: struct containing fee estimate
+    func estimateFeeV1(call: StarknetCall, nonce: Felt, simulationFlags: Set<StarknetSimulationFlagForEstimateFee>) async throws -> StarknetFeeEstimate {
```

Comment:
The number of overloaded functions is getting a little bit out of hand.

We could make `defaultSimulationFlagsForEstimateFee` public and then use it as default argument.
We could also replace `simulationFlags: Set<StarknetSimulationFlagForEstimateFee>` with a single `skipValidate: Bool` parameter. Currently, `SKIP_VALIDATE` is the only supported flag and I wouldn't count on new ones in a foreseeable future.

I decided not to deviate from `Provider.simulateTransactions`, since set approach is used there. But with just a single flag, it seems a bit excessive.

Reply:
My thoughts exactly. Let's do that. 
We must also remember to cast whatever we decide about jvm interfaces to swift, as I'm afraid suffixes make them a bit too complex

Reply:
> Let's do that.

Sure, which one of the approaches do you think we should go for?

Reply:
XD

Sorry, I had a brain fart. I'd just use simple `skipValidate` flag.

Reply:
Just to be sure, are we talking about `skipValidate` Bool, or a flag passed as `[.skipValidate]`?

Reply:
I had a Bool in mind, but whatever suits you best

Reply:
Went for the Bool approach for now.